### PR TITLE
python312Packages.oddsprout: 0.1.0 -> 0.1.1

### PR DESCRIPTION
**Pull request by itepastra**
Relax the dependency on `dahlia`, but this broke the import since

`dahlia` became `Dahlia`, so I made a patch to fix it.

Tested whether the binary didn't give errors.

ZHF: #352882

am waiting on https://github.com/trag1c/oddsprout/pull/4 to finish this

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
Please file an upstream PR and use `fetchpatch2`.

File: pkgs/development/python-modules/oddsprout/default.nix [hunk 1]
```diff
@@ -19,8 +20,19 @@ buildPythonPackage rec {
     hash = "sha256-k5/mBoW4PxGUbkwaZyHgS3MGI4533V/nNoGqEg+VXpM=";
   };
 
+  patches = [
+    (fetchpatch2 {
+      name = "update-dahlia-dependency";
```
Comment (dotlambda) [inline pkgs/development/python-modules/oddsprout/default.nix:25]:
```suggestion
      name = "update-dahlia-dependency.patch";
```

File: pkgs/development/python-modules/oddsprout/default.nix [hunk 2]
```diff
@@ -19,8 +20,19 @@ buildPythonPackage rec {
     hash = "sha256-k5/mBoW4PxGUbkwaZyHgS3MGI4533V/nNoGqEg+VXpM=";
   };
 
+  patches = [
+    (fetchpatch2 {
```
Comment (dotlambda) [inline pkgs/development/python-modules/oddsprout/default.nix:24]:
```suggestion
    # https://github.com/trag1c/oddsprout/pull/4
    (fetchpatch2 {
```

**Comment by dotlambda** [comment]
Should we mark this as a draft until the upstream PR is accepted?

**Comment by itepastra** [comment]
Yeah, I'm gonna make a better update upstream, we should just mark this broken for now until I get it updated upstream

**Comment by trag1c** [comment]
Just a heads-up: oddsprout 0.1.1 has been released :)

**Comment by itepastra** [comment]
@dotlambda, upstream got version bumped and I've updated the derivation as well. (thanks @trag1c for the quick reviews)

`nixpkgs-review` succeeds for both for me (but I forgot the --print-result flag and don't want to run it again)

**Comment by dotlambda** [comment]
@itepastra It might be worth adding yourself as a maintainer.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 354313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.oddsprout</li>
    <li>python311Packages.oddsprout.dist</li>
    <li>python312Packages.oddsprout</li>
    <li>python312Packages.oddsprout.dist</li>
  </ul>
</details>

**Comment by itepastra** [comment]
@dotlambda, should I swap the two commits or is that not needed

**Pull request merged**